src/components/compliance/resources/http.rs

```diff
@@ -0,0 +1,173 @@
+use bytes::BytesMut;
+use http::{Method, Request, Uri};
+use hyper::{Body, Client};
+use tokio::sync::mpsc;
+use tokio_util::codec::Encoder;
+use vector_core::event::Event;
+
+use crate::components::compliance::sync::{Configured, ExternalResourceCoordinator, WaitHandle};
+
+use super::{ResourceCodec, ResourceDirection};
+
+/// An HTTP resource.
+pub struct HttpConfig {
+    uri: Uri,
+    method: Option<Method>,
+}
+
+impl HttpConfig {
+    pub const fn from_parts(uri: Uri, method: Option<Method>) -> Self {
+        Self { uri, method }
+    }
+
+    pub fn spawn_as_input(
```

Comment:
It seems like there maybe should be a trait for these resource definition enum variants, that defines the `spawn_as_input()` and `span_as_output()`  ?

Reply:
Marking as resolved as the RFC has been significantly reworked and this no longer appears relevant.